tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot

```diff
@@ -14,25 +14,100 @@
 
 *** Settings ***
 Documentation  Test 11-03 - Upgrade-InsecureRegistry
-Suite Setup  Install VIC with version to Test Server  7315  --insecure-registry 10.10.10.10:1234
-Suite Teardown  Clean up VIC Appliance And Local Binary
 Resource  ../../resources/Util.robot
 
 *** Keywords ***
-Get host and path from guestinfo
-    ${host}=  Run  govc vm.info -e -json %{VCH-NAME} | jq -c '.VirtualMachines[0].Config.ExtraConfig[] | select(.Key | . and contains("guestinfo.vice./registry/insecure_registries|0/Host")) .Value'
-    ${path}=  Run  govc vm.info -e -json %{VCH-NAME} | jq -c '.VirtualMachines[0].Config.ExtraConfig[] | select(.Key | . and contains("guestinfo.vice./registry/insecure_registries|0/Path")) .Value'
-    [Return]  ${host}  ${path}
+Setup Test Environment
+    [arguments]  ${insecure_registry}
+    ${handle}  ${docker_daemon_pid}=  Start Docker Daemon Locally  --insecure-registry ${insecure_registry}
+    Setup VCH And Registry  ${insecure_registry}
+    [Return]  ${handle}  ${docker_daemon_pid}
+
+Add Project On Registry
+    [Tags]  secret
+    [Arguments]  ${registry_ip}  ${protocol}  ${user}=admin  ${password}=%{TEST_PASSWORD}
+    # Harbor API: https://github.com/vmware/harbor/blob/master/docs/swagger.yaml
+    Run Keyword If  '${protocol}' == 'https'  Set Test Variable  ${curl_option}  --insecure
+    Run Keyword If  '${protocol}' == 'http'  Set Test Variable  ${curl_option}  ${EMPTY}
+    :FOR  ${i}  IN RANGE  12
+    \   Sleep  10s
+    \   ${rc}=  Run And Return Rc  curl ${curl_option} -u ${user}:${password} -H "Content-Type: application/json" -X POST -d '{"project_name": "test","public": 1}' ${protocol}://${registry_ip}/api/projects
+    \   Continue For Loop If  '${rc}' != '0'
+    \   Run Keyword If  '${rc}' == '0'  Set Test Variable  ${pass}  1
+    \   Exit For Loop If  '${rc}' == '0'
```

Comment:
1. yes, if practice, I've seen that even though harbor ui is up, curl may still fail, getting a 'connection refused' or something. i feel like this is because that part of harbor is initialized after ui;
2. like above, we cannot use return from keyword if because that assumes we can always succeed in this for loop. but sometimes, for example, ci server is really slow, it could use out of all loop without succeeding, in that case, test should fail here instead of silently going through every loop and returning.